src/cart/crt_iv.c

```diff
@@ -1579,23 +1684,34 @@ crt_iv_fetch(crt_iv_namespace_t ivns, uint32_t class_id,
 
 	if (ivns_internal == NULL) {
 		D_ERROR("Invalid ivns\n");
+		CRT_EXIT();
 		return -DER_NONEXIST;
 	}
 
+	/* Get group name space interal operations */
 	iv_ops = crt_iv_ops_get(ivns_internal, class_id);
 	if (iv_ops == NULL) {
 		D_ERROR("Failed to get iv_ops for class_id = %d\n", class_id);
 		/* ADDREF done above in lookup */
 		IVNS_DECREF(ivns_internal);
+		CRT_EXIT();
 		return -DER_INVAL;
 	}
 
+	/* Get local version for latter comparison.  */
+	/* MUST obtain before getting root rank      */
+	D_RWLOCK_RDLOCK(&ivns_internal->cii_grp_priv->gp_rwlock);
+	grp_ver_entry = ivns_internal->cii_grp_priv->gp_membs_ver;
+
+	/* Get iv_key root rank */
```

Comment:
we are holding a lock during a callback?

Reply:
lock is held on purpose as the group version that we have must be corresponding to the root rank that we've retrieved from the caller.

Reply:
The lock is not held during a call back.  Need to have group version with associated root rank.  Modified proceeding comments.